presto-hive/src/test/java/com/facebook/presto/hive/TestHiveNativeWorkersQueries.java

```diff
@@ -0,0 +1,108 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.tpch.TpchTable.NATION;
+
+public class TestHiveNativeWorkersQueries
+        extends AbstractTestQueryFramework
+{
+    protected TestHiveNativeWorkersQueries()
+    {
+        super(TestHiveNativeWorkersQueries::createQueryRunner);
+    }
+
+    private static QueryRunner createQueryRunner()
+            throws Exception
+    {
+        String prestoServerPath = System.getenv("PRESTO_SERVER");
+        checkArgument(prestoServerPath != null,
+                "Path to native Presto server binary is missing. Specify it using PRESTO_SERVER environment variable.");
+
+        String baseDataDir = System.getenv("DATA_DIR");
+        checkArgument(baseDataDir != null,
+                "Path to data files is missing. Specify it using DATA_DIR environment variable");
+
+        // Make TPC-H tables in DWRF format using Java-based workers
+        HiveQueryRunner.createQueryRunner(ImmutableList.of(NATION),
+                ImmutableMap.of(),
+                "sql-standard",
+                ImmutableMap.of("hive.storage-format", "DWRF"),
+                Optional.of(Paths.get(baseDataDir))).close();
+
+        // Make query runner with native workers for tests
+        DistributedQueryRunner queryRunner = HiveQueryRunner.createQueryRunner(ImmutableList.of(NATION),
+                ImmutableMap.of("optimizer.optimize-hash-generation", "false"),
+                ImmutableMap.of(),
+                "sql-standard",
+                ImmutableMap.of(),
+                Optional.of(1),
+                Optional.of(Paths.get(baseDataDir)),
+                Optional.of((workerIndex, discoveryUrl) -> {
+                    try {
+                        if (workerIndex == 0) {
+                            // Write discovery URL to /tmp/config.properties
+                            Files.write(Paths.get("/tmp/config.properties"),
```

Comment:
Rather than presuming the existence of `/tmp`, can we just use `Files.createTempDirectory(prefix);`?  We could also tag the prefix with the test class name of this test (for discoverability, in case it accidentally creates too many local temp directories).